### PR TITLE
Use enum for refresh rate switching setting - allow choosing between scaling on device or on tv

### LINKED ISSUE #1646: Resolution Switching doesn't work anymore
### Describe the bug

1. Playback was excellent(?).
2. User asked for option to not resolution switch because their device is attempting to be a better upscaler: https://github.com/jellyfin/jellyfin-androidtv/issues/1482
3. Proposal outright removing resolution switching by @mueslimak3r: https://github.com/jellyfin/jellyfin-androidtv/pull/1484
4. Resolution switching doesn't work anymore.
5. @nielsvanvelzen asked for me to create a new issue.

Can we please revert this commit? See the log below for the behaviour that doesn't make sense. It looks like there's an issue with selecting the proper refresh rate as well.

### Logs

```shell
04-25 00:04:44.678 10445 10445 D PlaybackController: trying to find display mode for video: 1920x1080 @29.97003fps
04-25 00:04:44.679 10445 10445 D PlaybackController: qualifying display mode: 1920x1080 @59.94fps
04-25 00:04:44.680 10445 10445 D PlaybackController: preferring mode: 1920x1080 @59.94fps
04-25 00:04:44.680 10445 10445 D PlaybackController: qualifying display mode: 3840x2160 @59.94fps
04-25 00:04:44.680 10445 10445 D PlaybackController: qualifying display mode: 3840x2160 @29.97fps
04-25 00:04:44.681 10445 10445 D PlaybackController: preferring mode: 3840x2160 @29.97fps
04-25 00:04:44.681 10445 10445 D PlaybackController: qualifying display mode: 1920x1080 @29.97fps
04-25 00:04:44.682 10445 10445 I PlaybackController: *** Best refresh mode is: 8 - 3840x2160/29.969999
04-25 00:04:44.682 10445 10445 I PlaybackController: *** Attempting to change refresh rate from 3/59.94
```

### Application version

v0.13.3

### Where did you install the app from?

Amazon Appstore

### Device information

Firestock Max

### Android version

AMZN

### Jellyfin server version

unstable

### PR DESCRIPTION
**Pull request by mueslimak3r**
**Changes**
* Changed refresh rate switching setting (boolean) to enum (`disabled`, `scale on tv`, `scale on device`) and added a migration that migrates an old value of `true` -> `scale on tv`. `scale on tv` is chosen because scaling on the device seems to only be useful for devices like Nvidia Shield, which has "AI upscaling".

* added some extra logging to `findBestDisplayMode()` so that every mode will be logged
* standardized the logging format for printing "resolution@refresh rate" in `findBestDisplayMode()`: `%dx%d@%f`

**Issues**
fixes #1646

**Notes**
My display config doesn't support refresh rate or resolution switching so someone else will need to test this.

### CODE REVIEW

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml
Comment (mueslimak3r) [reply app/src/main/res/values/strings.xml:14]: I made this lbl_ because "disabled" could be reused for other stuff. I can rename it if you prefer that

Comment (nielsvanvelzen) [reply app/src/main/res/values/strings.xml:14]: Might want to call it something like "state_disabled" in that case.

**Review by nielsvanvelzen — APPROVED** [review]

**Comment by KyleSanderson** [comment]
Any update on this critical regression?

**Comment by mueslimak3r** [comment]
> Any update on this critical regression?

Hi Kyle, as mentioned in the issue you opened and in this PR, these changes need to be tested before they are merged.
I don't have a display that supports refresh rate switching so I can't test them.
Are you able to test them? That would probably accelerate the review process.

For the apk generated by this PR, you'd test it with a 10.8 server. To test with a 10.7.7 server, you'd need to cherrypick or merge the changes from this PR into a v0.13.3 branch.

**Comment by KyleSanderson** [comment]
> > Any update on this critical regression?
> 
> Hi Kyle, as mentioned in the issue you opened and in this PR, these changes need to be tested before they are merged. I don't have a display that supports refresh rate switching so I can't test them. Are you able to test them? That would probably accelerate the review process.
> 
> For the apk generated by this PR, you'd test it with a 10.8 server. To test with a 10.7.7 server, you'd need to cherrypick or merge the changes from this PR into a v0.13.3 branch.

I don't have write access on this repository. Are you able to create the merge request against the stable branch so this can be reasonably tested if user feedback is critical to the merge.

**Comment by Arial-Z** [comment]
Done some testing result here
https://github.com/jellyfin/jellyfin-androidtv/issues/1482#issuecomment-1113219204

**Comment by mueslimak3r** [comment]
I'll check the code again regarding it doing 1080p 50hz instead of 4k 50hz when using `scale on device`

**Comment by mueslimak3r** [comment]
@Arial-Z can you test the 1080p 50Hz mode with the new changes?

**Comment by Arial-Z** [comment]
Fixed display is at 4K@50hz with scale on device

**Comment by KyleSanderson** [comment]
```
05-07 13:03:55.292 21977 21977 I PlaybackController: Playing back in native mode.
05-07 13:03:55.293 21977 21977 D PlaybackController: trying to find display mode for video: 3840x2160@23.976025
05-07 13:03:55.294 21977 21977 D PlaybackController: considering display mode: 1 - 1920x1080@60.000004
05-07 13:03:55.294 21977 21977 D PlaybackController: considering display mode: 2 - 1920x1080@59.939999
05-07 13:03:55.295 21977 21977 D PlaybackController: considering display mode: 39 - 3840x2160@59.939999
05-07 13:03:55.296 21977 21977 D PlaybackController: considering display mode: 40 - 1280x720@59.939999
05-07 13:03:55.296 21977 21977 D PlaybackController: considering display mode: 41 - 3840x2160@50.000000
05-07 13:03:55.297 21977 21977 D PlaybackController: considering display mode: 42 - 3840x2160@60.000004
05-07 13:03:55.297 21977 21977 D PlaybackController: considering display mode: 43 - 3840x2160@30.000000
05-07 13:03:55.298 21977 21977 D PlaybackController: considering display mode: 44 - 3840x2160@29.969999
05-07 13:03:55.298 21977 21977 D PlaybackController: considering display mode: 45 - 3840x2160@25.000000
05-07 13:03:55.298 21977 21977 D PlaybackController: considering display mode: 46 - 3840x2160@24.000000
05-07 13:03:55.299 21977 21977 D PlaybackController: considering display mode: 47 - 3840x2160@23.976000
05-07 13:03:55.299 21977 21977 D PlaybackController: qualifying display mode: 47 - 3840x2160@23.976000
05-07 13:03:55.300 21977 21977 D PlaybackController: preferring mode: 47 - 3840x2160@23.976000
05-07 13:03:55.300 21977 21977 D PlaybackController: considering display mode: 48 - 1920x1080@50.000000
05-07 13:03:55.301 21977 21977 D PlaybackController: considering display mode: 49 - 1920x1080@29.969999
05-07 13:03:55.301 21977 21977 D PlaybackController: considering display mode: 50 - 1920x1080@23.976000
05-07 13:03:55.301 21977 21977 D PlaybackController: considering display mode: 51 - 1920x1080@24.000000
05-07 13:03:55.302 21977 21977 D PlaybackController: considering display mode: 52 - 1920x1080@30.000000
05-07 13:03:55.302 21977 21977 D PlaybackController: considering display mode: 53 - 1280x720@50.000000
05-07 13:03:55.303 21977 21977 D PlaybackController: considering display mode: 54 - 1280x720@60.000004
05-07 13:03:55.303 21977 21977 D PlaybackController: considering display mode: 55 - 720x576@50.000000
05-07 13:03:55.303 21977 21977 D PlaybackController: considering display mode: 56 - 720x480@60.000004
05-07 13:03:55.304 21977 21977 I PlaybackController: *** Best refresh mode is: 47 - 3840x2160/23.976000
05-07 13:03:55.304 21977 21977 I PlaybackController: *** Attempting to change refresh rate from: 39 - 3840x2160@59.939999
```

Thank you very much.

**Pull request merged**
